src/common/signalk/SKNMEAVisitor.cpp

```diff
@@ -27,19 +27,85 @@
   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
   THE SOFTWARE.
 */
-
+#include <Arduino.h>
 #include "SKUnits.h"
 #include "common/nmea/NMEASentenceBuilder.h"
 
 #include "SKNMEAVisitor.h"
+#include <KBoxLogging.h>
 
-void SKNMEAVisitor::visitSKElectricalBatteriesVoltage(const SKUpdate& u, const SKPath &p, const SKValue &v) {
-  NMEASentenceBuilder sb("II", "XDR", 4);
-  sb.setField(1, "V");
-  sb.setField(2, v.getNumberValue(), 2);
-  sb.setField(3, "V");
-  sb.setField(4, p.getIndex());
+// ***********************  Wind Speed and Angle  ************************
+//  Talker ID: WI Weather Instruments
+//             VW Velocity Sensor, Mechanical?
+//  also seen: IIVWR
+//
+//  NMEA0183  MWV Wind Speed and Angle  --> Deprecated
+//              1  2  3  4
+//              |  |  |  |
+//      $--MWV,x.x,a,x.x,a*hh
+//  1) Wind Angle, 0 to 360 degrees
+//  2) Reference, R = Relative, T = True
+//  3) Wind Speed
+//  4) Wind Speed Units, K/M/N
+//  5) Status, A = Data Valid
+//
+//      VWR Relative Wind Speed and Angle
+//              1  2  3  4  5  6  7  8
+//              |  |  |  |  |  |  |  |
+//      $--VWR,x.x,a,x.x,N,x.x,M,x.x,K*hh
+//  1) Wind direction magnitude in degrees
+//  2) Wind direction Left/Right of bow
+//  3) Speed
+//  4) N = Knots
+//  5) Speed
+//  6) M = Meters Per Second
+//  7) Speed
+//  8) K = Kilometers Per Hour
+// ************************************************************************
+void SKNMEAVisitor::visitSKEnvironmentWindAngleTrueWater(const SKUpdate &u, const SKPath &p, const SKValue &v) {
+
+  float WindAngle;
+  float WindSpeed;
+
+  WindAngle = SKRadToDeg(v.getNumberValue());
+
+  // WindAngle and WindSpeed should come together......
+  // TODO: one visit enough? which is coming first, Angle or Speed????
+
+  // Validation
+  if (WindAngle<0||WindAngle>360||!u.hasEnvironmentWindSpeedTrue()) return;
+
+  WindSpeed = u.getEnvironmentWindSpeedTrue();  // in m/s
+  DEBUG("WindSpeed: %f", WindSpeed);
+
+  NMEASentenceBuilder sb( "II", "MWV", 5);
+  sb.setField(1, WindAngle, 1 );
+  sb.setField(2, "R");
+  sb.setField(3, WindSpeed, 1 );
+  sb.setField(4, "M");
+  sb.setField(5, "A");
   _sentences.add(sb.toNMEA() + "\r\n");
+  //Serial.printf("%s\n", sb.toNMEA());
+
+  NMEASentenceBuilder sb1( "II", "VWR", 8);
```

Comment:
I think we should only send this when dealing with apparentWind, not for true wind.

My "bible" does not specify this:
http://www.catb.org/gpsd/NMEA.html#_vwr_relative_wind_speed_and_angle

But since it's called "relative wind speed and angle", I think it implies that this is for apparent. No?